http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -669,6 +669,80 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
             .raiseError[Unit](AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "permission denied"))
     } yield GetAppResponse.fromDbResult(app, Config.proxyConfig.proxyUrlBase)
 
+  // Autodeletion config logic requires a positive integer for threshold if enabled is true.
+  // For simplicity we can enforce this by rejecting the following scenarios:
+  // 1. The request includes a threshold which is invalid (0 or negative)
+  // 2. The request includes enabled=true but no threshold is provided, and the existing DB threshold is invalid
+  private def validateUpdateAppConfigRequest(req: UpdateAppConfigRequest, dbApp: App)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- as.ask
+
+    // 1. The request includes a threshold which is invalid (0 or negative)
+    invalidThresholdRequest = req.autodeleteThreshold.exists(_ <= 0)
+    _ <- F.raiseWhen(invalidThresholdRequest)(
+      BadRequestException("invalid value for autodeleteThreshold", Some(ctx.traceId))
+    )
+
+    // 2. The request includes enabled=true but no threshold is provided, and the existing DB threshold is invalid
+    wantEnableWithoutThresholdButDbInvalid = req.autodeleteEnabled.getOrElse(false) &&
+      req.autodeleteThreshold.isEmpty &&
+      dbApp.autodeleteThreshold.getOrElse(0) <= 0
+
+    _ <- F.raiseWhen(wantEnableWithoutThresholdButDbInvalid)(
+      BadRequestException(
+        "when enabling autodelete without passing an explicit autodeleteThreshold, the existing config must be valid",
+        Some(ctx.traceId)
+      )
+    )
```

Comment:
I feel like this logic was coded for CreateApp - is there any possibility in being able to reuse it at all?

Reply:
Personally ... I think it should be consistent (whatever it is). It seems / feels weird to have two different behaviors.

Talking it out here ...
If the autodelete is `enabled`, then we NEED to ensure that the `threshold` is `> 0`.

Given that the `threshold` is only valid if it is _greater than zero_, should we just restrict it to always being greater than zero?

Reply:
Agreed.  I found a nice way to do this.

Final logic for both create and update:
* reject threshold <=0 if present
* if enabled=true, reject if a threshold is not present (threshold value check is covered by previous step)